uom/common/defines.h

```diff
@@ -0,0 +1,35 @@
+// Copyright @2023 UOM project. All rights reserved.
+
+#pragma once
+
+#define CONST_FIELD(type, name)                \
+ public:                                       \
+  const type& name() const { return name##_; } \
+                                               \
+ private:                                      \
```

Comment:
If we write code according to google coding style, `CONST_FIELD ` will be used in private block, like
```c++
private:
  CONST_FIELD(int, count) = 0;
```

It will not influence anything.